client/utils/date-time.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { dateI18n } from '@wordpress/date';
+
+type DateTimeFormat = string | null;
+
+interface FormatDateTimeOptions {
+	includeTime?: boolean; // Whether to include time in the formatted string (defaults to true)
+	useGmt?: boolean; // Whether to display the time in GMT/UTC (defaults to false)
+	separator?: string; // Separator between date and time (defaults to ' / ')
+	customFormat?: DateTimeFormat; // Custom format to use instead of WordPress settings
+}
```

Comment:
If we add these comments as type annotations above, code editors (or VSCode at least) will show these on hover.

<img width="631" alt="image" src="https://github.com/user-attachments/assets/69c39606-0740-4ecc-840b-580b2bace80b">


```suggestion
interface FormatDateTimeOptions {
	/** Whether to include time in the formatted string (defaults to true) */
	includeTime?: boolean;
	/** Whether to display the time in GMT/UTC (defaults to false) */
	useGmt?: boolean;
	/** Separator between date and time (defaults to ' / ') */
	separator?: string;
	/** Custom format to use instead of WordPress settings, e.g. 'Y-m-d H:i' */
	customFormat?: DateTimeFormat;
}
```

This may help future devs understand `useGmt` – it wasn't immediately clear to me if it means the input is GMT or the output is GMT.

Reply:
> If we add these comments as type annotations above, code editors (or VSCode at least) will show these on hover.
That's a great tip, thanks for sharing. I'm following your suggestion here [298fe08](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/298fe08b0483b7ac6379167317853b6cae8a319d).

>This may help future devs understand useGmt – it wasn't immediately clear to me if it means the input is GMT or the output is GMT.
I agree; it wasn’t immediately clear to me either. Initially, I kept it because we were using it in certain places, apparently due to potential incompatibility with WordPress 5.4. However, after reconsidering, I’ve decided to remove it for two reasons:

Setting useGmt to true overrides the user’s settings, forcing the output to GMT. This goes against the purpose of this pull request, which aims to respect user preferences.
We’re no longer constrained by the WordPress 5.4 support policy, as the current major version is WordPress 6.7.
It's worth mentioning that removing useGmt: true caused some test snapshots to fail. To resolve this, I mocked the dateI18n function to use the GMT timezone, ensuring consistency in the tests.

I believe removing this parameter simplifies the code and improves clarity, but I’d love to hear your thoughts if you have any concerns or alternative suggestions.

Reply:
I was going to suggest _renaming_ `useGmt` so it's clearer what it does (e.g. `forceDisplayAsUtc` … but removing the option sounds even better 🚀

Reply:
Yeh good point, perhaps we can have a separate function for store TZ and GMT TZ, but both follow the date formatting/locale defined in WP settings.

This is similar to the separate `@wordpress/date` functions [`datei18n()`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n) for store TZ and [`gmdatei18n()`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#gmdatei18n) for GMT/UTC TZ.

Reply:
Thanks for your helpful comments, @haszari and @Jinksi!

If I understand correctly, you're suggesting that it would be beneficial to allow specifying the format as either the store's timezone (TZ) or UTC.  I’ve added a timezone option, leveraging the fact that the [datei18n()](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n) function supports it.

By default, this option is `undefined`, which means the utility functions will use the store's timezone. However, this also provides the flexibility to override the store's timezone by explicitly setting a different one (UTC or any other), should a need arise. While we don’t currently have a specific use case for this, it seemed like a simple enhancement.

You can review the changes in this commit: [933422e](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/933422e00040f48a13487650adc0c1615f8aa8af).